widgets/text/text_test.go

```diff
@@ -809,6 +809,112 @@ func TestTextDraws(t *testing.T) {
 				return ft
 			},
 		},
+		{
+			desc:   "tests maxContent length being applied - multiline",
+			canvas: image.Rect(0, 0, 10, 3),
+			opts: []Option{
+				MaxContent(10),
+				RollContent(),
+			},
+			writes: func(widget *Text) error {
+				return widget.Write("line0\nline1\nline2\nline3\nline4")
+			},
+			want: func(size image.Point) *faketerm.Terminal {
+				ft := faketerm.MustNew(size)
+				c := testcanvas.MustNew(ft.Area())
+				// \n still counts as a chacter in the string length
+				testdraw.MustText(c, "ine3", image.Point{0, 0})
+				testdraw.MustText(c, "line4", image.Point{0, 1})
+				testcanvas.MustApply(c, ft)
+				return ft
+			},
+		},
+		{
+			desc:   "tests maxContent exact length of 5",
```

Comment:
We should also consider what happens with unicode characters that are multi-cell and add a test case where the max content "cuts a rune in half" to verify that Termdash doesn't attempt to print half of a wide rune, but correctly skips it.

Unicode never fails to provide additional entertainment.

Reply:
I believe this is covered in the test `tests maxContent partial bufffer replacement`, as it has to truncate based on a character not the data (hence starting with 你 for the result). I was using this as a test case for this as I've never worked with Go for unicode before!

If I get the time I'll make some of the adjustments later in the week, otherwise happy if you want to merge and correct as well.